### ISSUE TITLE
Feature Request: Touchscreen Keypad for mode selection, expiry date, and quantities

### ISSUE THREAD

**Issue by cyberjacob** [Forceu/barcodebuddy#66]
Would be useful for small devices utilising `screen.php` to be able to change the scanning mode (purchase/consume), enter the expiry dates for scanned products and enter specific quantities.

I'm happy to look into implementing this, though I'll leave it to you to decide how you want it to look/work first.

**Comment by Forceu** [comment]
Yes, that sounds like a very good idea! Maybe with a button to show/hide the input?

I would love to implement that, however I do not have a lot of time at the moment, so if feel free to do a pull request :) The easiest way would be probably to implement it completely in Javascript and use the API for communication

**Comment by cyberjacob** [comment]
If the user has been authenticated, will API calls work without an API key?

**Comment by Forceu** [comment]
Not add the moment, but I will add that today (tracked in #69)

**Comment by Forceu** [comment]
Well that was easier than expected, implemented in https://github.com/Forceu/barcodebuddy/commit/300edbd549bde3b25a13ba01d6d6327e8a0f9395

**Comment by Forceu** [comment]
Partly done in 57dba901ffda23b89fe6d2beee0e548eeeef22db.
Expiry date to do.

**Comment by phatair** [comment]
Expiry date would be great. 

At the moment I scan products with barcodebuddy and then I have to edit every product again in grocy to add the "best before" / expire date.

**Comment by duracell** [comment]
Yes, that would be really great, I'm really confused, that there is no workflow to scan and add expire dates after coming back from the shop or did I miss something? Someone mentioned a script with the API, but 1. I couldn't find anything in the doc and then "bestBeforeInDays" sounds like I need to convert the date to days.

Additionally, there is a bug in 3.0.1 that you can't edit any product =/
